### DISCUSSION TITLE
ring-mqtt-ha-addon setup

### DISCUSSION THREAD

**Discussion by nirsilathevarajah** [tsightler/ring-mqtt#409] [Support Q&A]
Hi, I have been installing the ring-mqtt-ha-addon multiple times. I have even tried older versions of HassOS but I can't get the 
ring-mqtt-ha-addon to work.
I am assuming that the ring integration does not need to be installed for this, as this add-on already accesses all this information.
This is the steps I have done, if anyone can let me know if I am missing anything or doing something wrong, that would be greatly appreciated!
1) Install HassOS
2) Install Mosquito Broker Add-on
3) Configure Mosquito Integration (adding username and password)
4) Add ring-mqtt-ha-addon (https://github.com/tsightler/ring-mqtt-ha-addon) but manually adding it to the addon store by clicking the 3 dots on right side and pasting this URL
5) Install Ring-MQTT with Video Streaming( Add-on from the Github repo)
6) Open the Web-UI and enter Ring username and password and continue the 2FA 
7) The Web-UI says the add on is connected
The following are the logs for Mosquitto broker and Ring MQTT
<img width="1029" alt="Screen Shot 2022-07-28 at 5 22 40 PM" src="https://user-images.githubusercontent.com/98358802/181640122-2445f7c6-ac90-47ab-b762-d1d35ad58ee8.png">
<img width="759" alt="Screen Shot 2022-07-28 at 5 23 01 PM" src="https://user-images.githubusercontent.com/98358802/181640129-0b0ee3ec-705d-47e9-985d-d0a30d3a4349.png">

This is my overview page: I was assuming my ring cameras would show up at this point!
<img width="1065" alt="image" src="https://user-images.githubusercontent.com/98358802/181641033-94be410c-9f1c-43ec-826f-7daf2ae27654.png">

**Comment by tsightler** [comment]
Please read the log message, especially the log message that is tagged "IMPORTANT" as it is telling you what you need to check.

**Reply by nirsilathevarajah** [reply]
I am not sure how to change this if none of the cameras are showing up

**Reply by nirsilathevarajah** [reply]
![image](https://user-images.githubusercontent.com/98358802/181659753-76287ad1-bd6f-4854-9927-1d0d8bacce4d.png)
this is in my configurations, I don't know what to add in the locations_id

**Comment by tsightler** [comment]
The cameras are not showing up because discovery is not finishing, it is hanging.  You need to use the Ring app to verify that you do not have any offline hub devices and, if you do, either bring them back online or remove them from the app.

**Comment by nirsilathevarajah** [comment]
So I had one device that was offline and I deleted it from my ring app amd double checked to make sure it was not there.
This is the ring-mqtt log:
<img width="1009" alt="image" src="https://user-images.githubusercontent.com/98358802/181785758-3ae5c584-5c6c-4dc9-bd1c-be269293284d.png">
But I am still unable to see anything on the overview.

**Reply by tsightler** [reply]
Did you restart the addon after you did this?  I see above some restarts of HA, but not the addon.  If you continue to have issues please provide a screenshot of the Devices tab in the Ring app.  You can blur out the device names or you can send them to me via gmail (same username as here).  In almost 4 years I've never seen a case of discovery hanging that was not caused by an offline alarm or smart lighting hub.
